packages/pwa-kit-dev/src/ssr/server/build-dev-server.js

```diff
@@ -132,38 +133,92 @@ export const DevServerMixin = {
      * @private
      */
     _setupExtensions(app, options) {
+        logger.info('Setting up extensions...')
+
         // TODO: support extensions options array syntax i.e. ['extension-a', {}]
         const extensions = options.mobify?.app?.extensions || []
+        logger.info('Extensions to load', {
+            namespace: 'DevServerMixin._setupExtensions',
+            additionalProperties: {extensions: extensions}
+        })
+
         app.__extensions = extensions || []
 
         extensions.forEach((extension) => {
+            logger.info(`Loading extension: ${extension}`)
+
             const setupServerFilePathBase = path.join(
                 options.projectDir,
                 'node_modules',
                 extension,
+                'src',
                 'setup-server'
             )
-
             let filePath
             if (fs.existsSync(`${setupServerFilePathBase}.ts`)) {
                 filePath = `${setupServerFilePathBase}.ts`
             } else if (fs.existsSync(`${setupServerFilePathBase}.js`)) {
                 filePath = `${setupServerFilePathBase}.js`
             } else {
+                logger.warn(`No setup-server file found for ${extension}. Skipping.`)
                 return
             }
 
-            const setupServer = tsx.require(filePath, __filename)
-            if (!setupServer.default) {
-                console.warn(`Extension ${extension} does not have a default export. Skipping.`)
+            let ExtensionClass
+            try {
+                ExtensionClass = tsx.require(filePath, __filename).default
+            } catch (e) {
+                logger.error(`Error loading extension ${extension}:`, {
+                    namespace: 'DevServerMixin._setupExtensions',
+                    additionalProperties: {error: e}
+                })
                 return
```

Comment:
In the other file (build-remote-server.js), the error is _thrown_. But in this file, the error is not thrown.

In this case, can we make both files do the same thing?